usaspending_api/data_load/generic_loaders.py

```diff
@@ -0,0 +1,67 @@
+from usaspending_api.data_load.data_load_helpers import setup_load_lists, setup_mass_load_lists
+
+
+def insert_recipient_locations(cursor, load_objects):
+    columns, values = setup_mass_load_lists(load_objects, "recipient_location")
+    recipient_location_sql = "INSERT INTO references_location {} VALUES {} RETURNING location_id;".format(
+        columns, values
+    )
+
+    cursor.execute(recipient_location_sql)
+    return [tup[0] for tup in cursor.fetchall()]
+
+
+def insert_recipients(cursor, load_objects):
+    columns, values = setup_mass_load_lists(load_objects, "legal_entity")
+    recipient_sql = "INSERT INTO legal_entity {} VALUES {} RETURNING legal_entity_id;".format(columns, values)
+
+    cursor.execute(recipient_sql)
+    return [tup[0] for tup in cursor.fetchall()]
+
+
+def insert_place_of_performance(cursor, load_objects):
+    columns, values = setup_mass_load_lists(load_objects, "place_of_performance_location")
```

Comment:
Why does loading new records into `references_location` require two functions? Apologies if the answer is clear with a little more code examination.

Reply:
what two functions? If you count helper functions, it takes way more than two. If you don't it should take just one.

Reply:
`insert_recipient_locations()` and `insert_place_of_performance()` are inserting new records into `references_location`.

Reply:
Ahh, I see. Those two functions have the same destination table, but retrieve their values from different part of the load object, and have return values that are used in different foreign keys. There is room for code reuse here.

Reply:
closing conversation as unresolved but no longer requested.